fauxfactory/factories/numbers.py

```diff
@@ -60,3 +60,29 @@ def gen_positive_integer():
     min_value = 0
 
     return gen_integer(min_value=min_value)
+
+
+def gen_octagonal(min_value=None, max_value=None):
+    """Return a random octagonal number based on the current platform.
+
+    :returns: A random hexadecimal number.
+    :rtype: str
+
+    """
+    return oct(gen_integer(
```

Comment:
What should happen if `None` is passed?

```python
>>> int(str(None), base=8)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 8: 'None'
```

Also, why the `str` to `int` casting? If incorrect data is passed to `gen_integer` then we should see an exception raised...